### PR TITLE
Drop `ArrayMap` for `Map`.

### PR DESCRIPTION
**Pull request by wtgtybhertgeghgtwtg**
**What kind of change does this PR introduce?**
Drops the `ArrayMap` class for native `Map`.

**Did you add tests for your changes?**
No.

**If relevant, link to documentation update:**

**Summary**
Drops the `ArrayMap` class for the native `Map`.  Proposed to solve #3083.

**Does this PR introduce a breaking change?**
The `clone` method of `dependencyFactories` and `dependencyTemplates` no longer exists.  It is not used by any of the plugins in this repo, but it may be used by plugins outside of this repo.

Change
```
const newMap = dependencyTemplates.clone();
```
to
```
const newMap = new Map(dependencyTemplates);
```
**Other information**

### CODE REVIEW

File: lib/ArrayMap.js [hunk 1]
```diff
@@ -1,57 +0,0 @@
...
-				this.values[i] = value;
-				return this;
-			}
-		}
-		this.keys.push(key);
-		this.values.push(value);
-		return this;
-	}
-
-	remove(key) {
```
Comment (Kovensky) [inline lib/ArrayMap.js]: Need to check if anything was calling `.remove`. The native `Map` prototype method is `delete`.

Comment (wtgtybhertgeghgtwtg) [inline lib/ArrayMap.js]: Nothing in this repo seems to use `remove` or `clone`.

**Comment by TheLarkInn** [comment]
AppVeyor is currently down per https://twitter.com/appveyor/status/816518305770971137. Going to merge this.

**Comment by TheLarkInn** [comment]
Thank you very much!

**Pull request merged**